### PR TITLE
Remove WIP define for iothub client

### PR DESCRIPTION
**Pull request by jebrando**
# Description
Removes the WIP define that was used for a feature of AMQP methods

### CODE REVIEW

**Comment by BertKleewein** [comment]
> # Copyright (c) Microsoft. All rights reserved.

I assume that leaving these 2 scripts in the repo also means that we're leaving the Jenkins jobs that runs these, right?  Why not remove the jobs and remove these files?

---
Refers to: jenkins/linux_wip.sh:2 in 4999ee6. [](commit_id = 4999ee648bddbe2d4a60c6575cb8eb7568e6be39, deletion_comment = False)

File: dps_client/adapters/iothub_security_sas.c [hunk 1]
```diff
@@ -469,7 +469,7 @@ BUFFER_HANDLE iothub_security_sas_sign_data(SECURITY_DEVICE_HANDLE handle, const
         BYTE data_signature[1024];
         BYTE* data_copy = (unsigned char*)data;
         /* Codes_SRS_SECURE_DEVICE_TPM_07_021: [ iothub_security_sas_sign_data shall call into the tpm to hash the supplied data value. ] */
-        uint32_t sign_len = SignData(&handle->tpm_device, &NullPwSession, data_copy, data_len, data_signature, sizeof(data_signature) );
+        uint32_t sign_len = SignData(&handle->tpm_device, &NullPwSession, data_copy, (uint32_t)data_len, data_signature, (uint32_t)sizeof(data_signature) );
```
Comment (zolvarga) [inline dps_client/adapters/iothub_security_sas.c]: >(uint32_t)data_len, data_signature, (uint32_t)sizeof(data_signature) [](start = 85, length = 68)

I assume this change is not related to removing the WIP

Comment (BertKleewein) [inline dps_client/adapters/iothub_security_sas.c]: >(uint32_t) [](start = 85, length = 10)

Aargh.  I hate these kinds of casts.  Going from size_t to uint32_t might decrease or increase precision.  Can we update SignData to accept a size_t instead?  (I'm not looking to expand work scope -- just venting).

File: dps_client/src/dps_sasl_tpm.c [hunk 2]
```diff
@@ -124,7 +124,7 @@ static int construct_send_data(SASL_TPM_INSTANCE* sasl_tpm_info, unsigned char*
     }
     else
     {
-        payload_data_length = data_len + 1;
```
Comment (BertKleewein) [inline dps_client/src/dps_sasl_tpm.c]: >data_len + 1 [](start = 30, length = 12)

Are we concerned about overflow bugs here?  If we are, the cast complicates things.

File: dps_client/adapters/iothub_security_sas.c (hunk 1 above)
Comment (BertKleewein) [reply dps_client/adapters/iothub_security_sas.c]: +1.  Also adding soapbox comment about having 2 unrelated changes in a single commit and how that can complicate source code management.

---
In reply to: [138970052](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138970052) [](ancestors = 138970052)

Comment (jebrando) [reply dps_client/adapters/iothub_security_sas.c]: this is MSR security code,  I don't know the implication of doing this since size does matter in this code.

---
In reply to: [138970748](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138970748) [](ancestors = 138970748)

Comment (jebrando) [reply dps_client/adapters/iothub_security_sas.c]: Yea, yea.  Ok, you have made your point, I'll move this into to commits.

---
In reply to: [138972637](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138972637) [](ancestors = 138972637,138970052)

File: dps_client/src/dps_sasl_tpm.c (hunk 2 above)
Comment (jebrando) [reply dps_client/src/dps_sasl_tpm.c]: On 64 bit system the compiler requires a cast.

---
In reply to: [138971318](https://github.com/Azure/azure-iot-sdk-c/pull/231#discussion_r138971318) [](ancestors = 138971318)

**Comment by jebrando** [comment]
> # Copyright (c) Microsoft. All rights reserved.

probably you are correct.

---
In reply to: [329564064](https://github.com/Azure/azure-iot-sdk-c/pull/231#issuecomment-329564064) [](ancestors = 329564064)

---
Refers to: jenkins/linux_wip.sh:2 in 4999ee6. [](commit_id = 4999ee648bddbe2d4a60c6575cb8eb7568e6be39, deletion_comment = False)
